autorest/codegen/models/code_model.py

```diff
@@ -233,6 +234,15 @@ def _add_properties_from_inheritance(self) -> None:
         for schema in self.schemas.values():
             schema.properties = CodeModel._add_properties_from_inheritance_helper(schema, schema.properties)
 
+    def base_url_method_signature(self, async_mode: bool) -> str:
+        if async_mode:
+            if not self.base_url_default_value:
+                return "base_url: str,"
+            return "base_url: Optional[str] = None,"
```

Comment:
That seems complicated, why don't you have a parameter object, and just let it serialize as usual?

Reply:
Yeah, the reason I hesitated to make this a global parameter is bc it's on'y used in the client (we pass it to the pipeline client), while all global parameters are passed to the config as well (this is the current model). I did something to compensate for this in multiapi (since there are 3 params that are client-specific, don't get passed to the config). I didn't initially for this bc it's just one variable, but you're right I should create this as a paremter and do something like "service client specific variable"